lib/unifex/unifex_cnode.ex

```diff
@@ -0,0 +1,156 @@
+defmodule Unifex.UnifexCNode do
+  @doc """
+  Wraps Bundlex.CNode functionalities, in due to support specific Unifex's CNode behaviours
+  """
+
+  require Bundlex.CNode
+
+  @enforce_keys [:server, :node]
+  defstruct @enforce_keys
+
+  @type t :: %__MODULE__{
+          server: pid,
+          node: node
+        }
+
+  @type on_start_t :: {:ok, t} | {:error, :spawn_cnode | :connect_to_cnode}
+
+  @doc """
+  Casts specific for Bundlex Api structs returned from
+  Bundlex.CNode.start/1 or Bundlex.CNode.start_link/1 to the analogous structs
+  in Unifex.UnifexCNode API or vice versa
+  """
+  def cast_on_start_t({:ok, %Bundlex.CNode{} = bundex_cnode}) do
+    {:ok, cast_cnode(bundex_cnode)}
+  end
+
+  def cast_on_start_t({:ok, %__MODULE__{} = unifex_cnode}) do
+    {:ok, cast_cnode(unifex_cnode)}
+  end
+
+  def cast_on_start_t(on_start) do
+    on_start
+  end
+
+  @doc """
+  Casts Bundlex.CNode struct to Unifex.UnifexCNode struct or vice versa
+  """
+  def cast_cnode(%Bundlex.CNode{server: server, node: node}) do
```

Comment:
cast in Erlang world usually means something different, I'd rename it to something like `convert` or `upgrade`